janitor/functions.py

```diff
@@ -3815,3 +3816,67 @@ def jitter(
     df[dest_column_name] = result
 
     return df
+
+
+@pf.register_dataframe_method
+def sort_naturally(
+    df: pd.DataFrame, column_name: str, **natsorted_kwargs
+) -> pd.DataFrame:
+    """
+    Sort an DataFrame by a column using "natural" sorting.
+
+    To accomplish this, we do a natural sort
+    on the unique values that are present in the dataframe.
+    Then, we reconstitute the entire dataframe
+    in the naturally sorted order.
+
+    Natural sorting is provided by the Python package natsort_.
+
+    .. _natsort: https://natsort.readthedocs.io/en/master/index.html
+
+    All keyword arguments to natsort should be provided
+    after the column name to sort by is provided.
+    They are passed through to the ``natsorted`` function.
+
+    Functional usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor as jn
+
+        df = pd.DataFrame(...)
+
+        df = jn.functions.sort_naturally(
+            df=df,
+            column_name='alphanumeric_column',
+        )
+
+    Method chaining usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor
+
+        df = pd.DataFrame(...)
+
+        df = df.jitter(
+            column_name='alphanumeric_column',
+        )
+
+
+    This function returns a completely new dataframe object,
+    so attributes on the old dataframe are not preserved.
+
+    :param df: A pandas DataFrame.
+    :param column_name: The column on which natural sorting should take place.
+    :param natsorted_kwargs: Keyword arguments to be passed
+        to natsort's ``natsorted`` function.
+    """
+    sorted_values = natsorted(df[column_name].unique(), **natsorted_kwargs)
```

Comment:
Would something like this work instead?

```python
new_order = index_natsorted(df[column_name], **natsorted_kwargs)
return df.iloc[new_order, :]
```

Reply:
Let me try. I might have missed out on index_natsorted.

Reply:
Actually, what you have is better :smile:.